jwcrypto/jwk.py

```diff
@@ -482,6 +482,23 @@ class JWKSet(set):
 
     Inherits for the standard 'set' bultin type.
     """
+    def __init__(self, keyset=None):
+        if isinstance(keyset, str):
+            # Assume that this was a reimport
+            super(JWKSet, self).__init__()
+
+            try:
+                keyset = json_decode(keyset)['keys']
+            except:
+                raise InvalidJWKValue()
+
+            for jwk in keyset:
+                self.add(JWK(**jwk))
+
+        elif keyset is None:
+            super(JWKSet, self).__init__()
+        else:
+            super(JWKSet, self).__init__(keyset)
```

Comment:
What will this do?
Seems to me we should handle only None or a string that can be json_decode()d, not sure what good it makes to pass another unspecified value to the set() init ?